client/my-sites/invites/controller.js

```diff
@@ -14,10 +14,16 @@ import titleActions from 'lib/screen-title/actions';
 import InviteAccept from 'my-sites/invites/invite-accept';
 import { setSection } from 'state/ui/actions';
 import { renderWithReduxStore } from 'lib/react-helpers';
-import { acceptedNotice, getRedirectAfterAccept } from 'my-sites/invites/utils';
-import { successNotice } from 'state/notices/actions';
+import { getRedirectAfterAccept } from 'my-sites/invites/utils';
+import { acceptInvite as acceptInviteAction } from 'lib/invites/actions';
 
 export function acceptInvite( context ) {
+	const acceptedInvite = store.get( 'invite_accepted' );
+	if ( acceptedInvite && acceptedInvite.loggedOut ) {
+		acceptInviteAction( acceptedInvite, page.bind( null, getRedirectAfterAccept( acceptedInvite ) ) )( context.store.dispatch );
```

Comment:
We probably need to clear the `invite_accepted` from `localStorage` within this condition.

```
store.remove( 'invite_accepted' );
```